src/control/server/mgmt_svc.go

```diff
@@ -197,6 +206,17 @@ func (svc *mgmtSvc) Join(ctx context.Context, req *mgmtpb.JoinReq) (*mgmtpb.Join
 		return nil, errors.Wrap(err, "unmarshal Join response")
 	}
 
+	// if join successful, record membership
+	if resp.GetStatus() == 0 && resp.GetState() == mgmtpb.JoinResp_IN {
+		svc.mutex.Lock()
+		svc.members = append(svc.members, &SystemMember{
+			Addr: req.GetAddr(), Uuid: req.GetUuid(), Rank: resp.GetRank(),
+		})
+		svc.mutex.Unlock()
+
+		svc.log.Debugf("MgmtSvc.members: %+v\n", svc.members)
```

Comment:
I realize that this is just scaffolding, but I should point out that there is still a data race possible here. The append is done under lock, but this read is not. At a minimum this could result in test failures due to provoking the race detector. Is it worth holding up this PR? Probably not. I'll leave that as your call. If you want to fix this later that's fine with me.

As you develop future iterations of this feature, you might want to consider creating accessor methods for modifying/requesting the membership list. You can contain the locking to those methods for safe access. Another thing worth pointing out is that there may be logic which requires the entire membership list to be locked for certain operations, so it might make sense to move the locking to a new type which wraps the membership list with synchronized access.

Reply:
yes will fix up and iterate on future PRs, this is the first of a few